src/main/java/net/hypixel/nerdbot/generator/StringColorParser.java

```diff
@@ -9,310 +8,310 @@
+            int nextShortcut = description.indexOf("%%", charIndex + 1);
+            int nextNewLine = description.indexOf("\\n", charIndex + 1);
+            int nextPotentialMCColor = description.indexOf("&", charIndex + 1);
 
-                    int colorCheck = 36; // An extra buffer so we don't wrap colors
-                    boolean newLineFlag = true;
-                    for (int i = charIndex; i < charIndex + (colorCheck - lineLength); i++) {
-                        if (i + 1 > description.length()) {
-                            newLineFlag = false;
-                            break;
-                        }
-
-                        if (description.charAt(i) == ' ') {
-                            newLineFlag = false;
-                            break;
-                        }
-                        if (description.charAt(i) == '%' && description.charAt(i + 1) == '%') {
-                            colorCheck += 2;
-
-                            // Let's see if there's a color here. We'll check if it's valid later.
-                            for (int j = i + 2; j < description.length(); j++) {
-                                if (j + 2 <= description.length() && description.charAt(j) == '%' && description.charAt(j + 1) == '%') {
-                                    break;
-                                }
-                                colorCheck++;
-                            }
-                        }
-                    }
+            // finding the nearest place that can be line split at (spaces, %% \n or &)
+            int nearestSplit = nextSpace;
+            for (int item : new int[] {nextShortcut, nextNewLine, nextPotentialMCColor}) {
+                if (item != -1 && (nearestSplit == -1 || item < nearestSplit))
+                    nearestSplit = item;
+            }
+            // checks that there is a split that it can go to that isn't the end of the string
+            if (nearestSplit == -1) {
+                nearestSplit = description.length();
+            }
 
-                    if (newLineFlag) {
-                        parsed.add(currString.toString());
-                        currString.setLength(0);
-                        lineLength = 0;
-                    }
-                    continue;
-                }
+            String currentSubstring = description.substring(charIndex, nearestSplit);
+            if (lineLength + currentSubstring.length() > maxLineLength) {
+                // splitting the current string if it cannot fit onto a single line
+                if (currentSubstring.length() > maxLineLength)
+                    currentSubstring = currentSubstring.substring(0, maxLineLength + 1);
 
-                // EOL Parsing
-                if (lineLength > 35) {
-                    parsed.add(currString.toString());
-                    currString.setLength(0);
-                    lineLength = 0;
-                    continue;
+                this.createNewLine();
```

Comment:
I don't think the use of `this` is needed here unless I'm missing something?

Reply:
Just coding habits of mine and putting `this` everywhere